packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -228,10 +236,46 @@ abstract class CachingAssetBundle extends AssetBundle {
     return completer.future;
   }
 
+  @override
+  Future<dynamic> loadStructuredDataBinary(String key) async {
+    assert(key != null);
+    if (_standardMessageData.containsKey(key)) {
+      return _standardMessageData[key];
+    }
+
+    Completer<dynamic>? completer;
+    Future<dynamic>? result;
+    load(key).then<void>((ByteData value) {
```

Comment:
This could probably use `compute` to move the parsing to a separate thread

Reply:
I tried that with https://github.com/flutter/flutter/pull/113637/files#diff-1b6c749abe1192b78f5ddab5a43a95e382b6a7d9dc4fd675559d7e4b8800c8daR245-R251, but I get the following when trying to run the [benchmark](https://github.com/andrewkolos/flutter/blob/3739c09b256d2e34d09cf204dc3c413be17bcdf1/dev/benchmarks/microbenchmarks/lib/foundation/decode_and_parse_asset_manifest.dart#L16).

```
E/flutter ( 2582): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Binding has not yet been initialized.
E/flutter ( 2582): The "instance" getter on the ServicesBinding binding mixin is only available once that binding has been initialized.
E/flutter ( 2582): Typically, this is done by calling "WidgetsFlutterBinding.ensureInitialized()" or "runApp()" (the latter calls the former). Typically this call is done in the "void main()" method. The "ensureInitialized" method is idempotent; calling it multiple times is not harmful. After calling that method, the "instance" getter will return the binding.
E/flutter ( 2582): In a test, one can call "TestWidgetsFlutterBinding.ensureInitialized()" as the first line in the test's "main()" method to initialize the binding.
E/flutter ( 2582): If ServicesBinding is a custom binding mixin, there must also be a custom binding class, like WidgetsFlutterBinding, but that mixes in the selected binding, and that is the class that must be constructed before using the "instance" getter.
E/flutter ( 2582): #0      BindingBase.checkInstance.<anonymous closure> (package:flutter/src/foundation/binding.dart:287:9)
E/flutter ( 2582): #1      BindingBase.checkInstance (package:flutter/src/foundation/binding.dart:369:6)
E/flutter ( 2582): #2      ServicesBinding.instance (package:flutter/src/services/binding.dart:55:54)
E/flutter ( 2582): #3      PlatformAssetBundle.load (package:flutter/src/services/asset_bundle.dart:308:31)
E/flutter ( 2582): #4      CachingAssetBundle.thinkies (package:flutter/src/services/asset_bundle.dart:275:34)
E/flutter ( 2582): #5      _IsolateConfiguration.applyAndTime.<anonymous closure> (package:flutter/src/foundation/_isolates_io.dart:108:21)
E/flutter ( 2582): #6      Timeline.timeSync (dart:developer/timeline.dart:166:22)
E/flutter ( 2582): #7      _IsolateConfiguration.applyAndTime (package:flutter/src/foundation/_isolates_io.dart:106:21)
E/flutter ( 2582): #8      _spawn (package:flutter/src/foundation/_isolates_io.dart:127:67)
E/flutter ( 2582): #9      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:300:17)
E/flutter ( 2582): #10     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
E/flutter ( 2582): 
Reloaded 1 of 846 libraries in 1,117ms (compile: 27 ms, reload: 418 ms, reassemble: 403 ms).
Reloaded 451 of 846 libraries in 6,153ms (compile: 3433 ms, reload: 1512 ms, reassemble: 249 ms).
```

I tried adding `TestWidgetsFlutterBinding.ensureInitialized();` as the first statement in the test, but that remedy the issue anything. Have you seen this before?

Reply:
Don't load the data in the compute isolate, pass the data that is already loaded through the args. Just do the decode in the isolate.

Reply:
I found that doing this results in a significant slowdown ([code](https://github.com/andrewkolos/flutter/blob/7cd4ca08f40cc2d387572211aa1f969ad1a8ddb0/packages/flutter/lib/src/services/asset_bundle.dart#L246-L260)). The decoding only takes 2-3ms, perhaps the overhead from using another isolate is too much.

Reply:
Disregard. I just saw your new comment below.